draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2931,253 @@ processing the server's Certificate, CertificateVerify, and
+- A base key to be used to compute a MAC key.
+
+Based on these inputs, the messages then contain:
+
+Certificate
+: The certificate to be used for authentication and any
+supporting certificates in the chain.
+
+CertificateVerify
+: A signature over Handshake Context + Certificate.
+
+Finished
+: A MAC over Handshake Context + Certificate + CertificateVerify using
+  a MAC key derived from the base key.
+{:br}
+
+Because the CertificateVerify signs the Handshake Context +
+Certificate and the Finished MACs the Handshake Context + Certificate
++ CertificateVerify, this is mostly equivalent keeping a running hash
+of the handshake messages (exactly so in the pure 1-RTT cases). Note,
+however, the subsequent post-handshake authentications do not include
+each other, just the messages through the end of the main handshake.
+
+The following table defines the Handshake Context and MAC Key
+for each scenario:
+
+~~~~
+Mode             Handshake Context                        Base Key
+----             -----------------                        --------
+0-RTT            ClientHello + ServerConfiguration        xSS
+                             + Server Certificate
+                             + CertificateRequest
+                            
+1-RTT (Server)   ClientHello ... ServerConfiguration      master_secret
+                 
+1-RTT (Client)   ClientHello ... ServerFinished           master_secret
+
+Post-Handshake   ClientHello ... ClientFinished +         master_secret
+                 CertificateRequest
+~~~~
+Note 1: The ServerConfiguration, CertificateRequest, and Server Certificate
+in the 0-RTT case are the messages from the handshake where the
+ServerConfiguration was established.
+
+Note 2: The Handshake Context for the last three rows does not include
+any 0-RTT handshake messages, regardless of whether 0-RTT is used.
+
+
+
+####  Certificate
+
+When this message will be sent:
+
+> The server MUST send a Certificate message whenever the agreed-upon
+key exchange method uses certificates for authentication (this
+includes all key exchange methods defined in this document except PSK).
+
+> The client MUST send a Certificate message whenever the server has
+requested client authentication via a CertificateRequest message
+({{certificate-request}}) or when the EarlyDataIndication indicates
+client authentication. This message is only sent if the server
+requests a certificate. If no suitable certificate is available, the
+client MUST send a Certificate message containing no certificates.
 
-###  Server Certificate Verify
+Meaning of this message:
+
+> This message conveys the endpoint's certificate chain to the peer.
+
+> The certificate MUST be appropriate for the negotiated cipher suite's key
+exchange algorithm and any negotiated extensions.
+
+Structure of this message:
+
+%%% Authentication Messages
+       opaque ASN1Cert<1..2^24-1>;
+
+       struct {
+           ASN1Cert certificate_list<0..2^24-1>;
```

Comment:
Ack, I don't think that it matters, but CertificateVerify has a certain merit to it.  Of course, we should merge all the messages and then it will matter much less.